### PR TITLE
User auth

### PR DESCRIPTION
**Pull request by Deseteral**

### CODE REVIEW

File: service/src/main/kotlin/xyz/lebkuchenfm/external/security/Pbkdf2PasswordEncoder.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package xyz.lebkuchenfm.external.security
+
+import xyz.lebkuchenfm.domain.security.HashedPasswordHexEncoded
+import xyz.lebkuchenfm.domain.security.PasswordEncoder
+import javax.crypto.SecretKeyFactory
+import javax.crypto.spec.PBEKeySpec
+
+class Pbkdf2PasswordEncoder : PasswordEncoder {
+    private val factory = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA512")
+    private val iterationCount = 50000
```
Comment (gorzalczany) [inline service/src/main/kotlin/xyz/lebkuchenfm/external/security/Pbkdf2PasswordEncoder.kt:10]: it is recommended to use 210,000 iterations.
https://cheatsheetseries.owasp.org/cheatsheets/Password_Storage_Cheat_Sheet.html#pbkdf2

File: service/src/main/kotlin/xyz/lebkuchenfm/Application.kt [hunk 2]
```diff
@@ -150,20 +158,14 @@ fun Application.module() {
...
-            eventStreamRouting(eventStream, playerStateSynchronizer)
-        }
-
-        // TODO: Remove me.
         authenticate("auth-session") {
             authenticate("auth-bearer") {
-                get("/auth-test") {
-                    call.respondText { "You are authenticated!" }
+                route("/api") {
+                    authRouting()
```
Comment (gorzalczany) [inline service/src/main/kotlin/xyz/lebkuchenfm/Application.kt:164]: im guessing that auth routing is not yet finished and we have to prepare desired login screen?

File: service/src/main/kotlin/xyz/lebkuchenfm/external/security/Pbkdf2PasswordEncoder.kt (hunk 1 above)
Comment (Deseteral) [reply service/src/main/kotlin/xyz/lebkuchenfm/external/security/Pbkdf2PasswordEncoder.kt:10]: We've used 50000 iterations in v2. Changing this value will break currently saved credentials - forcing everyone to reset their passwords.

Comment (gorzalczany) [reply service/src/main/kotlin/xyz/lebkuchenfm/external/security/Pbkdf2PasswordEncoder.kt:10]: Do we want to add it to known problems?

File: service/src/main/kotlin/xyz/lebkuchenfm/Application.kt (hunk 2 above)
Comment (Deseteral) [reply service/src/main/kotlin/xyz/lebkuchenfm/Application.kt:164]: Everything about routing is already done, including login screen which lives inside client-beta package.

**Review by gorzalczany — APPROVED** [review]

File: service/src/main/kotlin/xyz/lebkuchenfm/external/security/Pbkdf2PasswordEncoder.kt (hunk 1 above)
Comment (Deseteral) [reply service/src/main/kotlin/xyz/lebkuchenfm/external/security/Pbkdf2PasswordEncoder.kt:10]: It already is :)

**Pull request merged**